xdsl/context.py

```diff
@@ -287,7 +287,19 @@ def get_dialect(self, name: str) -> "Dialect":
             raise UnregisteredConstructException(f"Dialect {name} is not registered")
         return dialect
 
-    def get_optional_dialect(self, name: str) -> "Dialect | None":
+    def get_optional_dialect(
+        self, name: str, load_if_needed: bool = False
```

Comment:
Can you remove this variable, and always assume that this is true?
So that this copies how attributes and types work.